### ISSUE TITLE
plop-reader on Pocketbook Inkpad Pro 3 systematically crashes

### ISSUE THREAD

**Issue by pajd** [pmartin/plop-reader#107]
Hello,

Bought a Pocketbook Inkpad Pro 3 as plop-reader worked so well on my Pocketbook HD. The problem is that the app opens, but then immediately crash whenever I click on refresh (or anywhere on the screen as a matter of fact). Log pasted below.

Tried to add the / at the end of the URL, to no avail.

Any clue of what could be the issue? Many thanks in advance for any hint.

[2020:07:18 20:34:49][debug] Initializing the application
[2020:07:18 20:34:49][debug] URL in config doesn't end with a '/' => adding one
[2020:07:18 20:34:49][debug] Switching to mode 1
[2020:07:18 20:34:49][debug] Showing entries: page:1/0 (total:0)
[2020:07:18 20:34:50][debug] Showing entries: page:1/0 (total:0)
[2020:07:18 20:35:01][debug] Initializing the application
[2020:07:18 20:35:01][debug] URL in config doesn't end with a '/' => adding one
[2020:07:18 20:35:01][debug] Switching to mode 1
[2020:07:18 20:35:01][debug] Showing entries: page:1/0 (total:0)
[2020:07:18 20:35:02][debug] Showing entries: page:1/0 (total:0)

**Comment by FrapiFrance** [comment]
I suspect an issue with firmware.
Plop Reader worked nice on my PB632 with firmware 5 (vivlio french version), and now that I updated to 6.1.785 , it crashes like you, as soon as I click on menu, refresh or any article. browsing the article list is OK.

**Comment by pajd** [comment]
Thanks a lot. Not sure I can downgrade firmware, which I guess could be a solution. And this excellent app does not look like it is maintained anymore unfortunately...

**Comment by FrapiFrance** [comment]
Yes, it's a pity. The only alternative I found was using KOreader as wallabag client, not so nice (as each article is seen by the default epub reader as a new book in the library)

**Comment by pajd** [comment]
Many thanks! KOreader is an helpful alternative meanwhile indeed, but not nearly as convenient as Plop-reader...

**Comment by svemoe** [comment]
Fyi, I made it running on my PB628, I had the same problem.
You can check [my fork](https://github.com/svemoe/plop-reader).
I can also provide a build if you want to test.
Note, it sure is a hacky solution, I am not a programmer and also the documentation for InkView not very good.
I just tried to make it work again, as the app is no longer maintained.

**Comment by FrapiFrance** [comment]
Hi @svemoe  , I'm definitely interested in your build, don't know exactly how to compile the source.
Will try it on a P632 with firmware 6.1.900

**Comment by svemoe** [comment]
Hi @FrapiFrance, please test:
[Plop! reader.app.zip](https://github.com/pmartin/plop-reader/files/5750938/Plop.reader.app.zip)
Note: I also deactivated downloading of preview images, since it crashed when the images are png (jpeg worked fine).

**Comment by FrapiFrance** [comment]
> Hi @FrapiFrance, please test:
> [Plop! reader.app.zip](https://github.com/pmartin/plop-reader/files/5750938/Plop.reader.app.zip)

It works !

So Plop! Reader from svmoe's fork https://github.com/svemoe/plop-reader/tree/hotfix
is working on [a] P632 (Vivlio Touch HD plus in France) with 
firmware U632.6.1.900 publication date 20201001_123124 localisation WW 

Thanks a lot

Meanwhile, I tried to compile it from your "hotfixes" branch, but was still WIP

**Comment by svemoe** [comment]
Glad to hear :)
I think there are still a lot of things to polish in the app, but it's difficult when the main developer is not active.
For help on compiling for pocketbook, you can check out the repo [pbmartin/pocketbook-demo](https://github.com/pmartin/pocketbook-demo).
Note: I compiled against the version of SDK_481 from [blchinezu/pocketbook-sdk](https://github.com/blchinezu/pocketbook-sdk), which is for FW5.17. A newer SDK is not available.

**Comment by FrapiFrance** [comment]
I managed to compile with this SDK. and without debug info .

Thanks for showing me this.

**Comment by pajd** [comment]
It does work indeed! Great surprise for 2021! Many thanks @svemoe !

**Comment by FrapiFrance** [comment]
My fork without debug info :
https://github.com/FrapiFrance/plop-reader/tree/for_firmware_6_4

**Comment by kalon33** [comment]
@FrapiFrance I just saw your messages a few days ago. That works perfectly with latest Pocketbook firmware (with an Inkpad color). Thanks!

**Comment by FrapiFrance** [comment]
> @FrapiFrance I just saw your messages a few days ago. That works perfectly with latest Pocketbook firmware (with an Inkpad color). Thanks!

My pleasure.

Just note that since last autumn (was it 2 years ago ?) old ssl lib no longer work with let's encrypt and self-certified https, so if used with self-hosted wallabag, you MUST use http only, no ssl.
